### DISCUSSION TITLE
Can't display unicode 🖴 symbol character

### DISCUSSION THREAD

**Discussion by f3rn4nd0-c354r** [brndnmtthws/conky#1350] [Q&A] [answered]
Hello!
I know this is probably a problem in the font, but I tried a couple of fonts anf nothing seems to work, so I would appreciate some help.

I used the original configuration just as an exemple:

```
conky.config = {
    alignment = 'top_left',
    background = false,
    border_width = 1,
    cpu_avg_samples = 2,
    default_color = 'white',
    default_outline_color = 'white',
    default_shade_color = 'white',
    double_buffer = true,
    draw_borders = false,
    draw_graph_borders = true,
    draw_outline = false,
    draw_shades = false,
    extra_newline = false,
    font = 'DejaVu Sans Mono:size=12',
    gap_x = 60,
    gap_y = 60,
    minimum_height = 5,
    minimum_width = 5,
    net_avg_samples = 2,
    no_buffers = true,
    out_to_console = false,
    out_to_ncurses = false,
    out_to_stderr = false,
    out_to_x = true,
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'desktop',
    show_graph_range = false,
    show_graph_scale = false,
    stippled_borders = 0,
    update_interval = 1.0,
    uppercase = false,
    use_spacer = 'none',
    use_xft = true,
}

conky.text = [[
${color grey}Info:$color ${scroll 32 Conky $conky_version - $sysname $nodename $kernel $machine}
$hr
${color grey}Uptime:$color $uptime
${color grey}Frequency (in MHz):$color $freq
${color grey}Frequency (in GHz):$color $freq_g
${color grey}RAM Usage:$color $mem/$memmax - $memperc% ${membar 4}
${color grey}Swap Usage:$color $swap/$swapmax - $swapperc% ${swapbar 4}
${color grey}CPU Usage:$color $cpu% ${cpubar 4}
${color grey}Processes:$color $processes  ${color grey}Running:$color $running_processes
$hr
${color grey}File systems:
🖴 / $color${fs_used /}/${fs_size /} ${fs_bar 6 /}
${color grey}Networking:
Up:$color ${upspeed} ${color grey} - Down:$color ${downspeed}
$hr
${color grey}Name              PID     CPU%   MEM%
${color lightgrey} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color lightgrey} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color lightgrey} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color lightgrey} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
]]
```

The problem is the "hard disk" character. It always shows up as a box.

![image](https://user-images.githubusercontent.com/6833155/209659682-c0180b18-5849-451a-a7ca-9aefc885753c.png)

How can I fix this?
Thanks!

**Comment by brndnmtthws** [comment, marked as answer]
You need to use an emoji font specifically. Try `Noto Color`, for example.

**Reply by f3rn4nd0-c354r** [reply]
That way I get an error and Conky doesn't start

`X Error of failed request:  BadLength (poly request too large or internal Xlib length error)`


This was the change:

```
${color grey}File systems:
${font NotoColorEmoji}🖴 / $color${fs_used /}/${fs_size /} ${fs_bar 6 /}
```

I searched for this error and try the "unifont". But then I got a strange character again:

![image](https://user-images.githubusercontent.com/6833155/209718519-77bd6322-90a2-4f03-8fa1-76a16c040b07.png)

**Reply by brndnmtthws** [reply]
It should work in newer versions of libXft: https://gitlab.freedesktop.org/xorg/lib/libxft/-/merge_requests/1

When I say _should_, I mean it might not work at all if for whatever reason you don't have the necessary support/patches/release installed.

**Reply by brndnmtthws** [reply]
Looks like this commit is needed, too, which only landed 5 months ago it seems: https://gitlab.freedesktop.org/xorg/lib/libxft/-/commit/6a08279ee80dbd1a226592d7f18c862c2e4a7d3a

**Reply by f3rn4nd0-c354r** [reply]
Ok. Thanks for helping! I'll wait for Debian's upgrade.
It's currently on 2.3.2, so it might take a while...
But now I know it's a problem with the configuration!
